contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifier} from "src/periphery/cert/interfaces/IEigenDACertVerifier.sol";
+import {IEigenDACertVerifierRouter} from "src/periphery/cert/interfaces/IEigenDACertVerifierRouter.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgradeable {
+    mapping(uint32 => address) public certVerifiers;
+    uint32[] public certVerifierABNs;
+
+    event CertVerifierAdded(uint32 indexed abn, address indexed certVerifier);
+
+    error ABNNotInFuture(uint32 abn);
+    error ABNNotGreaterThanLast(uint32 abn);
+    error InvalidCertLength();
+    error NoCertVerifierAvailable();
+    error NoCertVerifierFound(uint32 rbn);
+
+    /// IEigenDACertVerifierRouter ///
+
+    function checkDACert(bytes calldata certBytes) external view returns (uint8) {
+        return IEigenDACertVerifier(getCertVerifierAt(_getRBN(certBytes))).checkDACert(certBytes);
+    }
+
+    function getCertVerifierAt(uint32 rbn) public view returns (address) {
+        return certVerifiers[_findClosestRegisteredABN(rbn)];
+    }
+
+    /// ADMIN ///
+
+    function initialize(address _initialOwner) external initializer {
+        _transferOwnership(_initialOwner);
+    }
+
+    function addCertVerifier(uint32 abn, address certVerifier) external onlyOwner {
```

Comment:
I think most / all of these functions would benefit from explicit documentation. What do the input parameters represent? What are the restrictions imposed by the functions on their values?

Reply:
Documentation is now in the interfaces.